### PR TITLE
Adding support for a post_filter #47

### PR DESCRIPTION
**Pull request by mxkh**
Adding support for a post_filter #47

### CODE REVIEW

**Comment by beowulfenator** [comment]
Thanks. I will review this and the other commit shotly.

**Comment by mxkh** [comment]
@beowulfenator on my local machine unit tests run without falling. elasticsearch version 2.3.3

**Comment by polakv** [comment]
hi guys, what is the status of this feature? will it be available anytime soon? thx

File: Query.php [hunk 1]
```diff
@@ -165,6 +165,12 @@ class Query extends Component implements QueryInterface
      * @since 2.0.4
      */
     public $options = [];
+    /**
+     * @var string|array The post filter query for differentially filter search results and aggregations
+     * @see https://www.elastic.co/guide/en/elasticsearch/guide/current/_post_filter.html
+     * @since 2.0.5
+     */
+    public $postFilter = [];
```
Comment (cebe) [inline Query.php]: should  be default null, meaning it is not added if not specified. See `$filter`. Maybe even move it below `$filter`.

File: Query.php [hunk 2]
```diff
@@ -654,4 +660,15 @@ public function addOptions($options)
         return $this;
     }
 
+    /**
+     * Set the post filter query
+     * @param string|array $filter
+     * @return $this the query object itself
+     */
+    public function addPostFilter($filter)
```
Comment (cebe) [inline Query.php]: name should be `postFilter` as this method only sets it.

File: QueryBuilder.php [hunk 3]
```diff
@@ -370,12 +383,18 @@ private function buildHalfBoundedRangeCondition($operator, $operands)
...
+                $range_operator = 'lte';
+            } else {
+                if (in_array($operator, ['gt', '>'])) {
+                    $range_operator = 'gt';
+                } else {
+                    if (in_array($operator, ['lt', '<'])) {
+                        $range_operator = 'lt';
+                    }
+                }
+            }
```
Comment (cebe) [inline QueryBuilder.php]: this should not be changed, please revert this change.

File: tests/QueryBuilderTest.php [hunk 4]
```diff
@@ -34,9 +34,12 @@ public function setUp()
...
         $command = $this->getConnection()->createCommand();
-        $command->insert('yiitest', 'article', ['title' => 'I love yii!', 'weight' => 1, 'created_at' => '2010-01-10'], 1);
-        $command->insert('yiitest', 'article', ['title' => 'Symfony2 is another framework', 'weight' => 2, 'created_at' => '2010-01-15'], 2);
-        $command->insert('yiitest', 'article', ['title' => 'Yii2 out now!', 'weight' => 3, 'created_at' => '2010-01-20'], 3);
+        $command->insert('yiitest', 'article', ['title' => 'I love yii!', 'weight' => 1, 'created_at' => '2010-01-10'],
+            1);
+        $command->insert('yiitest', 'article',
+            ['title' => 'Symfony2 is another framework', 'weight' => 2, 'created_at' => '2010-01-15'], 2);
+        $command->insert('yiitest', 'article',
+            ['title' => 'Yii2 out now!', 'weight' => 3, 'created_at' => '2010-01-20'], 3);
```
Comment (cebe) [inline tests/QueryBuilderTest.php]: this should not be changed, please revert this change.

File: tests/QueryTest.php [hunk 5]
```diff
@@ -302,7 +308,8 @@ public function testBatch()
 
         //test each
         $query = new Query;
-        $query->from('yiitest', 'user')->limit(3)->orderBy(['name' => SORT_ASC])->indexBy('name')->options(['preference' => '_local']);
+        $query->from('yiitest',
+            'user')->limit(3)->orderBy(['name' => SORT_ASC])->indexBy('name')->options(['preference' => '_local']);
```
Comment (cebe) [inline tests/QueryTest.php]: this should not be changed, please revert this change.

File: tests/QueryTest.php [hunk 6]
```diff
@@ -320,7 +327,8 @@ public function testBatch()
 
         //test batch
         $query = new Query;
-        $query->from('yiitest', 'user')->limit(3)->orderBy(['name' => SORT_ASC])->indexBy('name')->options(['preference' => '_local']);
+        $query->from('yiitest',
+            'user')->limit(3)->orderBy(['name' => SORT_ASC])->indexBy('name')->options(['preference' => '_local']);
```
Comment (cebe) [inline tests/QueryTest.php]: this should not be changed, please revert this change.

**Comment by polakv** [comment]
Thanks @cebe. Can I somehow help with merging this pull request or I just need to wait for @mxkh to work in requested changes?

**Comment by cebe** [comment]
Merged manually, thanks!
